lib/gui/app/components/target-selector/target-selector-modal.tsx

```diff
@@ -0,0 +1,409 @@
+	hasListDriveImageCompatibilityStatus,
+	isDriveValid,
+	hasDriveImageCompatibilityStatus,
+} from '../../../../shared/drive-constraints';
+import { compatibility } from '../../../../shared/messages';
+import { bytesToClosestUnit } from '../../../../shared/units';
+import { getDrives, hasAvailableDrives } from '../../models/available-drives';
+import { getImage, getSelectedDrives } from '../../models/selection-state';
+import { store } from '../../models/store';
+import * as analytics from '../../modules/analytics';
+import { open as openExternal } from '../../os/open-external/services/open-external';
+import { Modal } from '../../styled-components';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+
+export interface DrivelistTarget extends DrivelistDrive {
+	displayName: string;
+	progress: number;
+	device: string;
+	link: string;
+	linkTitle: string;
+	linkMessage: string;
+	linkCTA: string;
+}
+
+/**
+ * @summary Get a drive's compatibility status object(s)
+ *
+ * @description
+ * Given a drive, return its compatibility status with the selected image,
+ * containing the status type (ERROR, WARNING), and accompanying
+ * status message.
+ */
+function getDriveStatuses(
+	drive: DrivelistTarget,
+): Array<{ type: number; message: string }> {
+	return getDriveImageCompatibilityStatuses(drive, getImage());
+}
+
+const TargetsTable = styled(({ refFn, ...props }) => {
+	return <BaseTable<DrivelistTarget> ref={refFn} {...props}></BaseTable>;
+})`
+	[data-display='table-head']
+		[data-display='table-row']
+		> [data-display='table-cell']:first-child {
+		padding-left: 15px;
+	}
+
+	[data-display='table-head']
+		[data-display='table-row']
+		> [data-display='table-cell'] {
+		padding: 6px 8px;
+		color: #2a506f;
+	}
+
+	[data-display='table-body']
+		> [data-display='table-row']
+		> [data-display='table-cell']:first-child {
+		padding-left: 15px;
+	}
+
+	[data-display='table-body']
+		> [data-display='table-row']
+		> [data-display='table-cell'] {
+		padding: 6px 8px;
+		color: #2a506f;
+	}
+`;
+
+interface DriverlessDrive {
+	link: string;
+	linkTitle: string;
+	linkMessage: string;
+}
+
+interface TargetStatus {
+	message: string;
+	type: number;
+}
+
+function badgeShadeFromStatus(status: string) {
+	switch (status) {
+		case compatibility.containsImage():
+			return 16;
+		case compatibility.system():
+			return 5;
+		default:
+			return 14;
+	}
+}
+
+function renderStatuses(statuses: TargetStatus[]) {
+	return _.map(statuses, (status) => {
+		const badgeShade = badgeShadeFromStatus(status.message);
+		return (
+			<Badge key={status.message} shade={badgeShade}>
+				{status.message}
+			</Badge>
+		);
+	});
+}
+
+const InitProgress = styled(
+	({
+		value,
+		...props
+	}: {
+		value: number;
+		props?: React.ProgressHTMLAttributes<Element>;
+	}) => {
+		return <progress max="100" value={value} {...props}></progress>;
+	},
+)`
+	/* Reset the default appearance */
+	-webkit-appearance: none;
```

Comment:
Do we still need the `-webkit` prefixes ?

Reply:
Not on this one, but we do on the rules below